liberty-ls/src/main/java/io/openliberty/tools/langserver/utils/PropertiesValidationResult.java

```diff
@@ -32,18 +38,79 @@ public PropertiesValidationResult(PropertiesEntryInstance entry) {
-            // ignoring case, check if value is valid
-            hasErrors = !validValues.stream().anyMatch(entry.getValue()::equalsIgnoreCase);
+        String value = entry.getValue();
+
+        if (ServerPropertyValues.usesPredefinedValues(textDocumentItem, property)) {
+            List<String> validValues = ServerPropertyValues.getValidValues(textDocumentItem, property);
+            if(ServerPropertyValues.isCaseSensitive(property)) {
+                // if case-sensitive, check if value is valid
+                hasErrors = !validValues.contains(value);
+            } else {
+                // ignoring case, check if value is valid
+                hasErrors = !validValues.stream().anyMatch(value::equalsIgnoreCase);
+            }
+            diagnosticType = ParserFileHelperUtil.isBootstrapPropertiesFile(textDocumentItem) ? 
+                                            "INVALID_PROPERTY_VALUE" : "INVALID_VARIABLE_VALUE";
+        } else if (ServerPropertyValues.usesIntegerRangeValue(property)) {
+            Range<Integer> range = ServerPropertyValues.getIntegerRange(property);
+            if (ServerPropertyValues.usesTimeUnit(property)) {
+                // for purgeMinTime - with or without unit (h or H) is acceptable.
+                try { // try parsing Integer if no units specified
+                    hasErrors = !range.contains(Integer.parseInt(value));
+                } catch (NumberFormatException e) { // contains unit (non-decimal char)
+                    // only accept integer followed by h or H
+                    Pattern integerAndTime = Pattern.compile("(^[0-9]+)([h|H]?)$");
+                    Matcher matcher = integerAndTime.matcher(value);
+                    if (matcher.find()) {
+                        try {
+                            int val = new BigInteger(matcher.group(1)).intValueExact();
+                            hasErrors = !range.contains(val);
+                        } catch (ArithmeticException ae) { // value too large for Integer
+                            hasErrors = true;
+                        }
+                    } else { // did not contain or start with integer
+                        hasErrors = true;
+                    }
+                }
+            } else {
+                try {
+                    hasErrors = !range.contains(Integer.parseInt(value));
+                } catch (NumberFormatException e) {
+                    hasErrors = true;
+                }
+            }
+            diagnosticType = ParserFileHelperUtil.isBootstrapPropertiesFile(textDocumentItem) ? 
+                                            "INVALID_PROPERTY_INTEGER_RANGE" : "INVALID_VARIABLE_INTEGER_RANGE";
+        } else if (ServerPropertyValues.usesPackageNames(property)) { // defines packages
+            // simple check for comma-delimited list of Java packages
+            Pattern packageList = Pattern.compile("^([a-z]+(\\.[a-z][a-z0-9]*)*)(,[a-z]+(\\.[a-z][a-z0-9]*)*)*$");
```

Comment:
According to this [page](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html), the underscore `_` is also allowed.